transport/tlscommon/tls_config.go

```diff
@@ -285,27 +286,13 @@ func makeVerifyConnection(cfg *TLSConfig) func(tls.ConnectionState) error {
 
 func makeVerifyServerConnection(cfg *TLSConfig) func(tls.ConnectionState) error {
 	switch cfg.Verification {
-	case VerifyFull:
-		return func(cs tls.ConnectionState) error {
-			if len(cs.PeerCertificates) == 0 {
-				if cfg.ClientAuth == tls.RequireAndVerifyClientCert {
-					return ErrMissingPeerCertificate
-				}
-				return nil
-			}
 
-			opts := x509.VerifyOptions{
-				Roots:         cfg.ClientCAs,
-				Intermediates: x509.NewCertPool(),
-				KeyUsages:     []x509.ExtKeyUsage{x509.ExtKeyUsageAny},
-			}
-			err := verifyCertsWithOpts(cs.PeerCertificates, cfg.CASha256, opts)
-			if err != nil {
-				return err
-			}
-			return verifyHostname(cs.PeerCertificates[0], cs.ServerName)
-		}
-	case VerifyCertificate:
+	// VerifyFull would attempt to match 'host' (c.ServerName) that is the host
+	// the client is trying to connect to with a DNS, IP or the CN from the
+	// client's certificate. Such validation, besides making no sense on the
+	// server side also causes errors as the client certificate usually does not
+	// contain a DNS, IP or CN matching the server's hostname.
+	case VerifyFull, VerifyCertificate:
```

Comment:
Conceptually this  sounds like an odd default behaviour for me, however I'm not an expert on mTLS.

My main question/concern here is: Is client certificates not containing a hostname a correct behaviour or is it a 'common, but technically wrong practice'?

Reply:
it's the correct behaviour, this check should not be there